src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java

```diff
@@ -223,17 +224,16 @@ public String getRemote() {
         GitHub github = Connector.connect(apiUri, credentials);
         try {
             if (credentials != null && !github.isCredentialValid()) {
-                listener.getLogger().format("Invalid scan credentials %s to connect to %s, skipping%n", CredentialsNameProvider.name(credentials), apiUri == null ? "github.com" : apiUri);
-                return;
+                String message = String.format("Invalid scan credentials %s to connect to %s, skipping%n", CredentialsNameProvider.name(credentials), apiUri == null ? "github.com" : apiUri);
+                throw new AbortException(message);
             }
             if (!github.isAnonymous()) {
                 listener.getLogger().format("Connecting to %s using %s%n", apiUri == null ? "github.com" : apiUri, CredentialsNameProvider.name(credentials));
             } else {
                 listener.getLogger().format("Connecting to %s with no credentials, anonymous access%n", apiUri == null ? "github.com" : apiUri);
             }
             if (repository == null || repository.isEmpty()) {
-                listener.getLogger().println("No repository selected, skip");
-                return;
+                throw new AbortException("No repository selected, skipping");
```

Comment:
The same than above https://github.com/jenkinsci/github-branch-source-plugin/pull/57#discussion_r63092478